test/e2e/disruption.go

```diff
@@ -42,63 +42,123 @@ var _ = framework.KubeDescribe("DisruptionController [Feature:PodDisruptionbudge
 	})
 
 	It("should create a PodDisruptionBudget", func() {
-		pdb := policy.PodDisruptionBudget{
+		createPodDisruptionBudgetOrDie(cs, ns, intstr.FromString("1%"))
+	})
+
+	It("should update PodDisruptionBudget status", func() {
+		createPodDisruptionBudgetOrDie(cs, ns, intstr.FromInt(2))
+
+		createPodsOrDie(cs, ns, 2)
+
+		err := wait.PollImmediate(framework.Poll, 60*time.Second, func() (bool, error) {
+			pdb, err := cs.Policy().PodDisruptionBudgets(ns).Get("foo")
+			if err != nil {
+				return false, err
+			}
+			return pdb.Status.PodDisruptionAllowed, nil
+		})
+		Expect(err).NotTo(HaveOccurred())
+
+	})
+
+	It("should allow an eviction when there is no PDB", func() {
+		createPodsOrDie(cs, ns, 1)
+
+		pod, err := cs.Pods(ns).Get("pod-0")
+		Expect(err).NotTo(HaveOccurred())
+
+		e := &policy.Eviction{
 			ObjectMeta: api.ObjectMeta{
-				Name:      "foo",
+				Name:      pod.Name,
 				Namespace: ns,
 			},
-			Spec: policy.PodDisruptionBudgetSpec{
-				Selector:     &unversioned.LabelSelector{MatchLabels: map[string]string{"foo": "bar"}},
-				MinAvailable: intstr.FromString("1%"),
-			},
 		}
-		_, err := cs.Policy().PodDisruptionBudgets(ns).Create(&pdb)
+
+		err = cs.Pods(ns).Evict(e)
 		Expect(err).NotTo(HaveOccurred())
 	})
 
-	It("should update PodDisruptionBudget status", func() {
-		pdb := policy.PodDisruptionBudget{
+	It("should not allow an eviction when too few pods", func() {
```

Comment:
Next test polls, presumably because condition starts out false and then becomes true. By the same reasoning, this test might pass because condition starts out false, and it would have eventually turned true, but you'll never detect that. So I think you should insert some kind of delay here (I'm not sure how long) to give it a chance to turn true, before checking to make sure it is still false.